CodingStandard.md

```diff
@@ -0,0 +1,1425 @@
+  - constexpr Methods and Functions: `lower_case`
+  - Non-const struct member variables: `lower_case`
+  - Non-const class member variables: `lower_case_` (incl. trailing undercore)
+  - Const member variables: `kCamelCase`
+  - Template type parameters: `CamelCase`
+  - Template value parameters: `kCamelCase`
+  - Namespaces (4 characters max) : `lower_case`
+
+
+- Get methods should be named for the variable they return, and set methods
+  should be the variable name prefixed with `set_`.
+
+- Abstract base classes (interfaces) should be prefixed with I, as in
+  `IContainer`.
+
+- Acronyms and abbreviations should be avoided, unless they are well-known and
+  commonly encountered throughout the project.
+
+- Acronyms in identifiers should be treated as words, for example,
+  `SidStarApproachRecord` as a class name and `sid_star_approach_record` as a
+  variable name.
+
+- Hungarian naming convention or any encoding of names to imply type information
+  should not be used.  This includes prefixing `p` for pointers, `m` for member
+  variables, `s` for static variables, etc.
+```
+[These are the standards we were using at UASC, and I'm not particularly attached
+to them. Specifically, I wouldn't mind having a convention that differentiates
+between public and private members and methods, and I definitely dislike the 
+trailing `_` they use for class members.  What do you guys like?]
+
+**Header Guards**
+
+All headers should have `#define` guards that match the file name.  The `#endif`
+should be followed by a comment indicating the symbol.  For example,
+some_file.h would be guarded as follows:
+
+```c++
+#ifndef SOME_FILE_H
+#define SOME_FILE_H
+
+   ...
+
+#endif // SOME_FILE_H
+```
+
+[The #pragma once Macro accomplishes this same goal, but it is <a 
+
+href="https://en.wikipedia.org/wiki/Pragma_once">none-standard</a>.]
+
+**Header File Layout**
+
+A typical .h file should look like this:
+
+```c++
+/*
+ *  (Copyright and File Header)
+ */
+#ifndef FOOS_FOO_H
+#define FOOS_FOO_H
+
+namespace foos {
+
+/*
+ *  (Class Header)
+ */
+struct Widget
+{
+    int x{};
+    int y{};
+};
+
+/*
+ *  (Class Header)
+ */
+class Foo
+{
+public:
+    Foo();
+    explicit Foo(int bar);
+    Foo(int bar, int id);
+    Foo(Foo& other) = delete;
+    Foo(Foo&& other) = delete;
+    Foo& operator=(Foo& other) = delete;
+    Foo& operator=(Foo other) = delete;
+    ~Foo();
+    int bar() const
+    {
+        return bar_;
+    }
+    void set_bar(int bar)
+    {
+        bar_ = bar;
+    }
+    int double_bar() const;
+
+private:
+    void compute_bar(int x);
+    int bar_;
+    int id_;
+    Widget widget_;
+}
+
+/*
+ *  (Routine Header)
+ */
+inline int Foo::double_bar() const
+{
+    return 2 * bar_;
+}
+
+} // namespace foos
+
+#endif // FOOS_FOO_H
+```
+
+Sections should be ordered public, protected, then private.
+
+Class members should be declared in this order:
+ - Using-declarations, typedefs and enums
```

Comment:
I dislike using, I think using explicit namespaces is clearer, and can avoid some ambiguity.

However I do like auto when appropriate.

Reply:
I like `using` as a replacement for typedef: `using IntegerType = int32_t`.